apps/Gruntfile.js

```diff
@@ -335,36 +355,127 @@ module.exports = function (grunt) {
+        districtDropdown: './src/code-studio/districtDropdown.js',
+        'levels/contract_match': './src/code-studio/levels/contract_match.jsx',
+        'levels/widget': './src/code-studio/levels/widget.js',
+        'levels/external': './src/code-studio/levels/external.js',
+        // put these entry points in arrays so that they can be required elsewhere
+        // https://github.com/webpack/webpack/issues/300
+        'levels/multi': ['./src/code-studio/levels/multi.js'],
+        'levels/textMatch': ['./src/code-studio/levels/textMatch.js'],
+        'levels/levelGroup': './src/code-studio/levels/levelGroup.js',
+        'levels/dialogHelper': './src/code-studio/levels/dialogHelper.js',
+        'initApp/initApp': './src/code-studio/initApp/initApp.js'
+      },
+      commonFile: 'code-studio-common',
+      provides: ['marked', 'react', 'react-dom', 'radium']
+    },
+
+    {
+      uniqueName: 'plc',
+      entries: {
+        plc: './src/code-studio/plc/plc.js'
+      },
+      provides: ['marked']
+    },
+
+    // Build embedVideo.js in its own step (skipping factor-bundle) so that
+    // we don't have to include the large code-studio-common file in the
+    // embedded video page, keeping it fairly lightweight.
+    // (I wonder how much more we could slim it down by removing jQuery!)
+    // @see embed.html.haml
+    {
+      uniqueName: 'embedVideo',
+      entries: {
+        embedVideo: './src/code-studio/embedVideo.js'
+      },
+      provides: ['jquery']
+    },
+
+    // embedBlocks.js is just React, the babel-polyfill, and a few other dependencies
+    // in a bundle to minimize the amound of stuff we need when loading blocks
+    // in an iframe.
+    {
+      uniqueName: 'embedBlocks',
+      entries: {
+        embedBlocks: './src/code-studio/embedBlocks.js'
+      },
+      provides: ['react', 'react-dom', 'radium']
+    },
+
+    {
+      uniqueName: 'makerlabDependencies',
+      entries: {
+        makerlab: './src/code-studio/makerlab/makerlabDependencies.js'
+      },
+      provides: ['johnny-five', 'playground-io', 'chrome-serialport']
+    },
+
+    {
+      uniqueName: 'pd',
+      entries: {
+        pd: './src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx'
+      }
+    }
+  ];
+
+  // Create a config for each of our bundles
+  function createConfigs(bundles, options) {
+    var minify = options.minify;
+    var watch = options.watch;
+
+    return bundles.map(function (bundle) {
+      var uniqueName = bundle.uniqueName;
+      var entries = bundle.entries;
+      var commonFile = bundle.commonFile;
+      var provides = bundle.provides;
+
+      return webpackConfig.create({
+        uniqueName: bundle.uniqueName,
+        output: path.resolve(__dirname, OUTPUT_DIR),
+        entries: entries,
+        commonFile: commonFile,
+        provides: provides,
+        minify: minify,
+        watch: watch,
+        piskelDevMode: PISKEL_DEVELOPMENT_MODE
+      });
+    });
```

Comment:
I think this is the reason why the build got slower. When your webpack config is an array of webpack configs, it's (unfortunately) like running webpack separately for each config. Depending on the performance hit, it's probably acceptable to get this merged in now and then go back to fix the performance.

I think ultimately what we want to do is just have a single webpack config object that generates all the bundles.